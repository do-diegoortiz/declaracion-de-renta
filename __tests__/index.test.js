import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import App from '../pages/index.js'

describe('initial', () => {
  test('first tests', () => {
    expect(true).toBe(true)
  })
})

describe('With Enzyme', () => {
  it('App shows right first parragraph', () => {
    const app = shallow(<App />)

    expect(app.find('p').first().text()).toEqual('Hay algunas cosas que puedes hacer para reducir legalmente el valor final a pagar por ingresos de trabajo del 2019. Pero las debes hacer antes del 31 de diciembre de 2019.')
  })
})

describe('With Snapshot Testing', () => {
  it('App shows "Hello world!"', () => {
    const component = renderer.create(<App />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})