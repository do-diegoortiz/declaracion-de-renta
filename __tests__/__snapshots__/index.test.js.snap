// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Hello world!" 1`] = `
<div>
  <h1>
    Optimiza la declaración de renta
  </h1>
  <p>
    Hay algunas cosas que puedes hacer para reducir legalmente el valor final a pagar por ingresos de trabajo del 2019. Pero las debes hacer antes del 31 de diciembre de 2019.
  </p>
  <h2>
    Ingresos
  </h2>
  <div>
    <form
      onSubmit={[Function]}
    >
      <h1>
        1
      </h1>
      <label
        htmlFor="contract"
      >
        Contrato
      </label>
      <select
        id="contract"
        name="contract"
        onChange={[Function]}
        value="nomina"
      >
        <option
          value="nomina"
        >
          Nómina
        </option>
        <option
          value="prestaciones"
        >
          Prestacion
        </option>
        <option
          value="contratista"
        >
          Contratista
        </option>
      </select>
      <input
        name="income"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        placeholder="($) Salario mensual en COP"
        required={true}
        type="text"
        value=""
      />
      <label
        htmlFor="de"
      >
        De
      </label>
      <input
        name="fromDate"
        onChange={[Function]}
        type="date"
      />
      <label
        htmlFor="hasta"
      >
        Hasta
      </label>
      <input
        defaultValue="2019-12-31"
        name="toDate"
        onChange={[Function]}
        type="date"
      />
      <input
        type="submit"
        value="Calcular"
      />
    </form>
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    Agregar Ingreso+
  </button>
  <div />
  <h2>
    Deducciones
  </h2>
  <p>
    Escriba en cada casilla el valor total que espera pagar en el año.
  </p>
  <form>
    <div>
      <label>
        Medicina Prepagada
      </label>
      <input
        name="prepaidMedicine"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        placeholder="$0"
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        Aportes Como Ind. a SS
      </label>
      <input
        name="indepSocialSecurity"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        placeholder="$0"
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        Cantidad Dependientes
      </label>
      <input
        name="dependants"
        onChange={[Function]}
        type="number"
        value={0}
      />
    </div>
    <div>
      <label>
        Donaciones
      </label>
      <input
        name="donations"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        placeholder="$0"
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        Aportes a Pens Voluntar.
      </label>
      <input
        name="voluntaryContributions"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        placeholder="$0"
        type="text"
        value=""
      />
    </div>
  </form>
</div>
`;
